packages/react-instantsearch-hooks-dom/src/ui/__tests__/Highlight.test.tsx

```diff
@@ -115,7 +133,37 @@ describe('Highlight', () => {
   test('forwards `className` and root props', () => {
     const { container } = render(
       <Highlight
-        baseClassName="ais-Highlight"
+        classNames={{
+          root: 'ROOT',
+          highlighted: 'HIGHLIGHTED',
+          nonHighlighted: 'NON-HIGHLIGHTED',
+          separator: 'SEPARATOR',
+        }}
+        parts={[]}
+        className="custom-root"
+        aria-hidden="true"
+      />
+    );
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <span
+          aria-hidden="true"
+          class="ROOT custom-root"
+        />
+      </div>
+    `);
+  });
+
+  test('accepts array of classes for classNames', () => {
+    const { container } = render(
+      <Highlight
+        classNames={{
+          root: ['ROOT', 'EXTRA', false],
```

Comment:
I can see why this is convenient, but couldn't this be done by calling `cx` on the passed classes instead?

```diff
-<Highlight classNames={{ root: ['ROOT', 'EXTRA', false] }} />
+<Highlight classNames={{ root: cx('ROOT', 'EXTRA', false) }} />
```

Reply:
I guess calling cx on two levels is fairly equivalent to flattening inside cx. Not sure what we prefer as the team?

Reply:
I'm good either way. But if we accept arrays, we need to add a test in the `cx` test suite.